src/containers/Application.js

```diff
@@ -312,9 +317,36 @@ class Application extends Component {
 			return Promise.reject(error);
 		});
 
-		// We want to remove networking indicator when we recieve the response
+		// We want to remove the networking indicator when we receive the response
 		axios.interceptors.response.use(function (response) {
+			// Call the `popNetworkingIndicator` method to remove the networking indicator (e.g., loading spinner)
 			that.popNetworkingIndicator();
+
+			// Check if the response content type is 'application/json' or starts with 'application/json'
+			if ((response.headers['content-type'] == 'application/json') || (response.headers['content-type']?.startsWith('application/json;'))) {
```

Comment:
@aringocode I suppose this should be just

```
response.headers['content-type']?.startsWith('application/json')
```

your proposed condition do kinda redundant checks, also its incorrect in this part `.startsWith('application/json;')` - this actually pass only if it starts with `application/json;` with the semicolon `;` at the end

Reply:
And how you handle `application/json-my-completely-different-type` ?